core/src/test/scala/integration/kafka/api/AdminClientIntegrationTest.scala

```diff
@@ -1819,6 +1819,123 @@ class AdminClientIntegrationTest extends IntegrationTestHarness with Logging {
       classOf[InvalidTopicException])
     client.close()
   }
+
+  @Test
+  def testDescribeConfigsForLog4jLogLevels(): Unit = {
```

Comment:
It isn't clear to me if:
1. We are testing returning a log level configuration set by log4j.properties file?
2. What would be the config source in that case

Reply:
That is what we are testing, yes. This is a bit of a sanity check to see it behaves as expected and also serves as a bit of test-documentation such that anybody reading the code can see a quick example of the API.

Regarding the config source, it should (and is, as tested) be the DYNAMIC_BROKER_LOGGER_CONFIG. I found this out during implementation and mentioned in the vote thread:
```
During implementation, I noticed a couple of holes in the KIP and did the following changes:
* add a new ConfigResource - DYNAMIC_BROKER_LOGGER_CONFIG
* DELETE operations in the IncrementalAlterConfigRequest will change the logger's level to the ROOT logger's currently-set level. (we previously thought we could return to the default - log4j does not seem to allow us to do that)
```

The key being I previously thought we had a way of knowing the default but we don't, therefore the server can't return a specific config source mentioning it's a default. Note that we can't infer the default from the root logger as that can be changed as well or the specific logger could be configured to use a non-default log level.

Reply:
So, I'm still not sure what will someone get if they try to describe the log level config for a broker that just started. Basically "null" or "unset"?

Reply:
`null`. They will always get `null` for a non-configured logger until they dynamically change it

Reply:
Changing it to return the root logger now